modules/AlphaWalletFoundation/AlphaWalletFoundation/EtherClient/TransactionSigning.swift

```diff
@@ -97,111 +112,107 @@ public struct EIP155Signer: TransactionSigner {
         }
     }
 
-    public func rlpEncodedHash(transaction: UnsignedTransaction) throws -> Data {
-        switch transaction.gasPrice {
-        case .legacy(let gasPrice):
-            return try rlpEncodeForLegacyGasPriceHash(transaction: transaction, legacyGasPrice: gasPrice)
-        case .eip1559(let maxFeePerGas, let maxPriorityFeePerGas):
-            return try rlpEncodeForEip1559Hash(transaction: transaction, maxFeePerGas: maxFeePerGas, maxPriorityFeePerGas: maxPriorityFeePerGas)
-        }
-    }
-
-    private func rlpEncodeForLegacyGasPriceHash(transaction: UnsignedTransaction, legacyGasPrice: BigUInt) throws -> Data {
+    static func rlpEncodeForEip1559Hash(transaction: UnsignedTransaction, maxFeePerGas: BigUInt, maxPriorityFeePerGas: BigUInt) throws -> Data {
         let values: [Any] = [
+            transaction.server.chainID,
             transaction.nonce,
-            legacyGasPrice,
+            maxPriorityFeePerGas,
+            maxFeePerGas,
```

Comment:
Aarrgh. Unfortunately the diff makes it look scary. The functions are only moved, not modified.